Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/SubmissionRepository.kt

```diff
@@ -71,52 +71,49 @@ class SubmissionRepository @Inject constructor(
         }
     }
 
-    // TODO this should be more UI agnostic
     fun refreshDeviceUIState(refreshTestResult: Boolean = true) {
-        var refresh = refreshTestResult
+        if (LocalData.submissionWasSuccessful()) {
```

Comment:
Why is this necessasry to fix the scrolling issue of the statistics?

I like the refactoring, but it would be better as this is part of an extra PR so it's easier to explicitly test these changes.

Reply:
Yes, this is part of the bug ticket. It runs through several states quickly on each resume, resulting in quick changes of the list items.

Reply:
Ok :+1:.

Would be nice to have a little bit more PR description :grin:. 
What/Why, and what needs explicit testing, e.g. run a test result through all possible test states and go back and forth between home screen and statistics info.